pandas/core/series.py

```diff
@@ -3635,6 +3635,59 @@ def reorder_levels(self, order):
         result.index = result.index.reorder_levels(order)
         return result
 
+    def explode(self) -> 'Series':
+        """
+        Create new Series expanding a list-like column.
+
+        .. versionadded:: 0.25.0
+
+        Returns
+        -------
+        exploded: Series
+
+        See Also
+        --------
+        Series.str.split: Split string values on specified separator.
+        Series.unstakc: Unstack, a.k.a. pivot, Series with MultiIndex to produce DataFrame.
+
+        Examples
+        --------
+        In [1]: s = pd.Series([[1, 2, 3], 'foo', [], [3, 4]])
+
+        In [2]: s
+        Out[2]:
+        0    [1, 2, 3]
+        1          foo
+        2           []
+        3       [3, 4]
+        dtype: object
+
+        In [3]: s.explode()
+        Out[3]:
+        0      1
+        0      2
+        0      3
+        1    foo
+        2    NaN
+        3      3
+        3      4
+        dtype: object
+
+        Notes
+        -----
+        This routine will explode list-likes including lists, tuples, Series, and np.ndarray.
+        The result dtype of the returned Series will always be object.
+        Scalars will be returned unchanged.
+        Empty list-likes will result in a np.nan for that row.
+        """
+        if not len(self):
```

Comment:
Does it make sense to also catch non-object dtype here and return a copy, as there should be nothing to explode?  The non-object dtype behavior is currently a bit inconsistent:
```python
In [2]: # raises 
   ...: pd.Series([1, 2, 3]).explode()
---------------------------------------------------------------------------
ValueError: Buffer dtype mismatch, expected 'Python object' but got 'long'

In [3]: # changes to object dtype 
   ...: pd.Series([1, 2, 3], dtype='Int64').explode()
Out[3]: 
0    1
1    2
2    3
dtype: object

In [4]: # raises 
   ...: pd.Series(pd.date_range('2019', periods=3)).explode()
---------------------------------------------------------------------------
ValueError: cannot include dtype 'M' in a buffer

In [5]: # keeps dtype 
   ...: pd.Series(pd.date_range('2019', periods=3, tz='UTC')).explode()
Out[5]: 
0   2019-01-01 00:00:00+00:00
1   2019-01-02 00:00:00+00:00
2   2019-01-03 00:00:00+00:00
dtype: datetime64[ns, UTC]
```